lib/dal/src/secret.rs

```diff
@@ -557,29 +601,65 @@ impl Secret {
     ) -> SecretResult<Self> {
         // Generate a new key and insert a new encrypted secret.
         let new_key = Self::generate_key(ctx, self.id)?;
+
+        // NOTE(nick): we do not clean up the existing encrypted secret yet.
         EncryptedSecret::insert(ctx, new_key, crypted, key_pair_pk, version, algorithm).await?;
 
         // Since we are updating encrypted contents, we have a new key and need to enqueue ourselves
         // into dependent values update.
         ctx.enqueue_dependent_values_update(vec![self.id]).await?;
 
-        // Now, update the key on the secret on the graph.
         self.modify(ctx, |s| {
-            s.key = new_key;
+            s.encrypted_secret_key = new_key;
             Ok(())
         })
         .await
     }
 
-    /// Modifies the [`Secret`] and persists modifications as applicable.
     async fn modify<L>(self, ctx: &DalContext, lambda: L) -> SecretResult<Self>
     where
         L: FnOnce(&mut Self) -> SecretResult<()>,
     {
         let mut secret = self;
 
+        // NOTE(nick): I don't love the current timestamp and actor system. These likely shouldn't
+        // be in the contents, but abstracted out into another service. Because of this, we have to
+        // manually ensure that the actor and timestamp information is correct, regardless of what
+        // the user passes in as the lambda.
         let before = SecretContentV1::from(secret.clone());
         lambda(&mut secret)?;
+        if before != SecretContentV1::from(secret.clone()) {
+            match ctx.history_actor() {
+                HistoryActor::SystemInit => {}
+                HistoryActor::User(id) => {
+                    secret.updated_by = Some(*id);
+                }
+            }
+            secret.timestamp.updated_at = Utc::now();
+        }
```

Comment:
Now, we update the history actor and "updated_at" timestamp every time that something has changed. This is a bonus benefit of the PR.